vm/src/stdlib/posix.rs

```diff
@@ -473,12 +473,138 @@ pub mod module {
                 sched_priority_repr.as_str()
             ))
         }
+
+        #[cfg(any(
+            target_os = "linux",
+            target_os = "netbsd",
+            target_os = "freebsd",
+            target_os = "android"
+        ))]
+        fn try_to_libc(&self, vm: &VirtualMachine) -> PyResult<libc::sched_param> {
+            let priority = self.sched_priority.clone();
+            let priority_type = priority.class().name();
+            let value = priority.downcast::<int::PyInt>().map_err(|_| {
+                vm.new_type_error(format!(
+                    "an integer is required (got type {})",
+                    priority_type
+                ))
+            })?;
+            let sched_priority = int::try_to_primitive(value.as_bigint(), vm)?;
+            Ok(libc::sched_param { sched_priority })
+        }
     }
 
+    #[derive(FromArgs)]
+    pub struct SchedParamArg {
+        #[pyarg(any)]
+        sched_priority: PyObjectRef,
+    }
     impl SlotConstructor for SchedParam {
-        type Args = SchedParam;
-        fn py_new(cls: PyTypeRef, sched_param: Self::Args, vm: &VirtualMachine) -> PyResult {
-            sched_param.into_pyresult_with_type(vm, cls)
+        type Args = SchedParamArg;
+        fn py_new(cls: PyTypeRef, arg: Self::Args, vm: &VirtualMachine) -> PyResult {
+            SchedParam {
+                sched_priority: arg.sched_priority,
+            }
+            .into_pyresult_with_type(vm, cls)
+        }
+    }
+
+    #[cfg(any(
+        target_os = "linux",
+        target_os = "netbsd",
+        target_os = "freebsd",
+        target_os = "android"
+    ))]
+    #[pyfunction]
+    fn sched_getscheduler(pid: libc::pid_t, vm: &VirtualMachine) -> PyResult<i32> {
+        let policy = unsafe { libc::sched_getscheduler(pid) };
+        if policy == -1 {
+            Err(errno_err(vm))
+        } else {
+            Ok(policy)
+        }
+    }
+
+    #[cfg(any(
+        target_os = "linux",
+        target_os = "netbsd",
+        target_os = "freebsd",
+        target_os = "android"
+    ))]
+    #[derive(FromArgs)]
+    struct SchedSetschedulerArgs {
+        #[pyarg(positional)]
+        pid: i32,
+        #[pyarg(positional)]
+        policy: i32,
+        #[pyarg(positional)]
+        sched_param_obj: PyRef<SchedParam>,
```

Comment:
```suggestion
        sched_param_obj: crate::PyRef<SchedParam>,
```

will remove the warning